pq-src/build.rs

```diff
@@ -274,113 +284,182 @@ fn main() {
         basic_build.flag("-fsanitize=address");
     }
 
-    let (libports_os, libcommon_os, libpq_os) = match target_os.as_str() {
+    // Add necessary defines
+    add_defines(&target_os, use_openssl, &mut basic_build);
+
+    // Collect files for compilation
+    let (libports, libcommon, libpq) = collect_sources(&target_os, use_openssl);
+
+    // Check if strchrnul and/or strsignal are supported, if so add defines for them
+    conditional_define(TEST_FOR_STRCHRNUL, "HAVE_STRCHRNUL", &mut basic_build);
+    conditional_define(TEST_FOR_STRSIGNAL, "HAVE_STRSIGNAL", &mut basic_build);
+
+    // Compile code into statically linked library
+    // Note: The compilation will output to $OUT_DIR/libpq.a
+    basic_build
+        .files(libports.iter().map(|f| port_path.join(f)))
+        .files(libcommon.iter().map(|f| common_path.join(f)))
+        .files(libpq.iter().map(|f| libpq_path.join(f)))
+        .compile("pq");
+
+    // Copy over relevant headers for next step(s)
+    copy_headers(
+        &libpq_path,
+        &psql_include_path,
+        &additional_includes_path,
+        &out,
+    );
```

Comment:
This step is required that dependencies can depend on the libpq version we build, which means this enables them to allow our bundled version as build dependency for their C(++) code. A crate that uses this is gdal-src: https://github.com/georust/gdal/blob/bd760a8e3ba7cedb22f07727712b642f40229a43/gdal-src/build.rs#L286

This needs the headers sorted out + that environment variable set (done via the ` println!("cargo:include={out}/include");` a few lines below

Reply:
I have improved the comments around copying of headers in e4557ce to make it clearer what's happening